tests/spec_constants/spec_constants_defined_various_ways.h

```diff
@@ -72,9 +74,12 @@ void perform_test(util::logger &log, const std::string &type_name,
       }
     });
   }
-  if (!check_equal_values(ref, result))
-    FAIL(log,
-         "case (" + case_hint + ") for " + type_name_string<T>::get(type_name));
+  if (!no_target_kernel) {
```

Comment:
I think you can make the code more readable by removing negation from the variable name and if-condition.

I.e. 
```c++
bool has_target_kernel = true;
...
        if (!k_bundle.has_kernel(kernelId)) {
          has_target_kernel = false;
...

  // Check results only if target kernel is available
  if (has_target_kernel && !check_equal_values(ref, result))
      FAIL(log, "case (" + case_hint + ") for " +
                    type_name_string<T>::get(type_name));
```